framework/src/base/Attributes.C

```diff
@@ -309,21 +309,38 @@ AttribPreIC::isEqual(const Attribute & other) const
   return isMatch(other);
 }
 
-void
-AttribPreAux::initFrom(const MooseObject * /*obj*/)
-{
-}
 bool
 AttribPreAux::isMatch(const Attribute & other) const
 {
   auto a = dynamic_cast<const AttribPreAux *>(&other);
-  return a && (a->_val == _val);
+  if (!a || a->_vals.size() < 1)
+    return false;
+  auto cond = a->_vals[0];
+  if (cond == Moose::ALL)
+    return true;
+
+  for (auto val : _vals)
+    if (val == Moose::ALL || val == cond)
+      return true;
+  return false;
```

Comment:
You've allowed _vals to be a vector, yet you aren't comparing all entries to all entries.  Either a vector is the wrong data structure, or you need to account for this.

Reply:
I'll note this new implementation of `AttribPreAux` is based entirely on the existing attribute `AttribExecOns`. They are now very similar classes. I am willing to change the design of this class, but it will then be different from `AttribExecOns` even though `AttribPreAux` and `AttribExecOns` have nearly the same functionality.

The only difference between `AttribExecOns` and `AttribPreAux`  now is that `AttribExecOns` can fill its `_vals` object by a call to `AttribExecOns::initFrom(const MooseObject * obj)`, but that won't work for `AttribPreAux` so instead, an additional constructor is added which can be used to fill `_vals` 

The reason that `_vals` is a vector is so that there only needs to be one data member. The attribute classes appear to be used in two ways, one way to store attributes of an object and in a different way when querying for objects. `_vals` is filled with more than one value only when calling the constructor that takes a vector.  This constructor is used only when calling `TheWarehouse.update()` method and the vector is filled with all exec flags which are relevant to the object passed to `update()`. The `_vals` vector is filled with only single value when calling `TheWarehouse.query` like in this block of code:
```c++
void
FEProblemBase::computeUserObjects(const ExecFlagType & type, const Moose::AuxGroup & group)
{
  TheWarehouse::Query query =
      theWarehouse().query().condition<AttribSystem>("UserObject").condition<AttribExecOns>(type);
  computeUserObjectsInternal(type, group, query);
}

void
FEProblemBase::computeUserObjectsInternal(const ExecFlagType & type,
                                          const Moose::AuxGroup & group,
                                          TheWarehouse::Query & query)
{
  if (group == Moose::PRE_IC)
    query.condition<AttribPreIC>(true);
  else if (group == Moose::PRE_AUX)
    query.condition<AttribPreAux>(type);
```